src/Grid/grid.jl

```diff
@@ -71,7 +82,19 @@ end
 abstract type AbstractGrid{dim} end
 
 """
+    Grid{dim, C<:AbstractCell, T<:Real} <: AbstractGrid}
+
 A `Grid` is a collection of `Cells` and `Node`s which covers the computational domain, together with Sets of cells, nodes and faces.
+
+# Fields
+- `cells::Vector{C}`: stores all cells of the grid
+- `nodes::Vector{Node{dim,T}}`: stores the `dim` dimensional nodes of the grid
+- `cellsets::Dict{String,Set{CellIndex}}`: maps a `String` key to a `Set` of cell ids
+- `nodesets::Dict{String,Set{Int}}`: maps a `String` key to a `Set` of global node ids
+- `facesets::Dict{String,Set{FaceIndex}}`: maps a `String` to a `Set` of `Tuple{Int,Int} (global_cell_id, local_face_id)`
```

Comment:
We should be more precise here on what these things are doing and not just describing the data structure.

I think we can even make a group comment here. i think something like "Helper structures to group entities by their name name." could be a start.